### PR TITLE
Feature: Build with --no-cache option in command line

### PR DESCRIPTION
**Pull request by 06rajesh**

### CODE REVIEW

File: bob/playbook/playbook.go [hunk 1]
```diff
@@ -392,8 +412,11 @@ func (p *Playbook) TaskCompleted(taskname string) (err error) {
 	errz.Fatal(err)
 
 	// TODO: use target hash?
-	err = p.pack(taskname, hashIn)
-	errz.Fatal(err)
+	// pack artifacts if only playbook disableArtifacts are set to default value (false)
+	if !p.disableArtifacts {
```
Comment (Equanox) [inline bob/playbook/playbook.go:416]: Things like this don't read nicely (double negation). The "--no-cache" option is very common but for internal variables avoid using negative variable names.
For ref: http://davej.io/2018/05/negative-naming.html

File: test/e2e/artifacts/artifacts_no_cache_test.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+			Expect(err).NotTo(HaveOccurred())
+
+			fs, err := os.ReadDir(artifactDir)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(len(fs)).To(Equal(0))
+		})
+
+		It("should rebuild and create no artifact for all task build with no-cache", func() {
+			ctx := context.Background()
+			err := bNoCache.Build(ctx, "all")
```
Comment (Equanox) [inline test/e2e/artifacts/artifacts_no_cache_test.go:64]: Use the fixed tasknames from the playground https://github.com/benchkram/bob/blob/f4e2c8dc3fbe872950e8a261191a0e84d0df00d8/bob/playground.go#L23

File: cli/cmd_root.go [hunk 3]
```diff
@@ -39,6 +39,7 @@ func init() {
 
 	// buildCmd
 	buildCmd.Flags().Bool("dummy", false, "Create a dummy bobfile")
+	buildCmd.Flags().Bool("no-cache", false, "Make true to disable save and load target from the cache")
```
Comment (Equanox) [inline cli/cmd_root.go:42]: "do not use cache"

**Comment by Equanox** [comment]
`bob run` should also accept a `--no-cache` option.

File: bob/playbook/playbook.go (hunk 1 above)
Comment (06rajesh) [reply bob/playbook/playbook.go]: renamed all the variable names with negation. ex: disableCache -> enableCaching, disableArtifacts -> enableArtifacts and also changed the logic accordingly.

File: test/e2e/artifacts/artifacts_no_cache_test.go (hunk 2 above)
Comment (06rajesh) [reply test/e2e/artifacts/artifacts_no_cache_test.go]: updated

File: cli/cmd_root.go (hunk 3 above)
Comment (06rajesh) [reply cli/cmd_root.go]: Updated the description to "Set to true to not use cache"

**Review by Equanox — APPROVED** [review]

**Pull request merged**